src/components/inline_edit/inline_edit_form.tsx

```diff
@@ -186,20 +223,21 @@ export const EuiInlineEditForm: FunctionComponent<EuiInlineEditFormProps> = ({
 
   const cancelInlineEdit = () => {
     setEditModeValue(readModeValue);
+    onCancel && onCancel(readModeValue as string);
     setIsEditing(false);
     requestAnimationFrame(() => readModeFocusRef.current?.focus());
   };
 
   const saveInlineEditValue = async () => {
     // If an onSave callback is present, and returns false, stay in edit mode
     if (onSave) {
-      const onSaveReturn = onSave(editModeValue);
+      const onSaveReturn = onSave(value as string);
       const awaitedReturn =
         onSaveReturn instanceof Promise ? await onSaveReturn : onSaveReturn;
       if (awaitedReturn === false) return;
     }
 
-    setReadModeValue(editModeValue);
+    setReadModeValue(value);
```

Comment:
While this works, I think I'd still prefer to keep it with the old `editModeValue` as it's more explicit as to what's happening and requires less cognitive overhead to follow
```suggestion
    setReadModeValue(editModeValue);
```